homeassistant/components/growatt_server/sensor.py

```diff
@@ -81,13 +83,11 @@ async def async_setup_entry(
         hass.config_entries.async_update_entry(config_entry, data=config)
 
     # Initialise the library with a random user id each time it is started,
-    # also extend the library's default identifier to include 'home-assistant'
-    api = growattServer.GrowattApi(
-        add_random_user_id=True,
-        agent_identifier=(
-            f"{growattServer.GrowattApi.agent_identifier} - home-assistant"
-        ),
+    rand_num = random.randint(1, 50)
+    random_identifier = "".join(
+        random.choices(string.ascii_uppercase + string.digits, k=rand_num)
     )
+    api = growattServer.GrowattApi(agent_identifier=random_identifier)
```

Comment:
Why can't we use the `add_random_user_id` parameter in the library?

Reply:
I rushed this yesterday trying to get a fix out and forgot about it.
I've since thought of a better way of doing it which will include the username rather than just a random string.
People that a desperate know how to 'hack' things to make things work, I'll improve this PR tomorrow and use the suggestion you made.

Reply:
Fixed in a far nicer way that uses the upstream library function better.